### PR TITLE
Gatewayd - Authentication Tests

### PR DESCRIPTION
**Pull request by okjodom**
Introduces gatewayd to the test infrastructure by implementing a shared `ILnRpcClient` for the fake lightning test fixture. Then,
- [x] use gatewayd in fixtures
- [x] apply existing authentication tests on gatewayd

### CODE REVIEW

File: fedimint-testing/src/ln/fixtures.rs [hunk 1]
```diff
@@ -78,13 +96,54 @@ impl LnRpc for FakeLightningTest {
...
 
-    async fn route_hints(&self) -> Result<Vec<RouteHint>, Error> {
+    async fn route_hints(&self) -> std::result::Result<Vec<RouteHint>, Error> {
         Ok(vec![RouteHint(vec![])])
     }
 }
+
+#[async_trait]
+impl ILnRpcClient for FakeLightningTest {
+    async fn pubkey(&self) -> Result<GetPubKeyResponse> {
```
[the remark this answers is not in the document]
Comment (okjodom) [reply fedimint-testing/src/ln/fixtures.rs:112]: I expect the use of `std::Result` to be removed from this file soon. Only impl of `LnRpc` uses it and that gets replaced by impl of `ILnRpcClient`. In that case, is the result still confusing?

File: gateway/ln-gateway/src/gatewayd/actor.rs [hunk 2]
```diff
@@ -104,16 +107,18 @@ impl GatewayActor {
...
+                    None => {
                         warn!("HTLC stream closed by service");
                         None
                     }
-                    Err(e) => {
-                        error!("HTLC stream closed with error: {:?}", e);
-                        None
-                    }
                 } {
                     if subscription.is_shutting_down() {
```
[the remark this answers is not in the document]
Comment (okjodom) [reply gateway/ln-gateway/src/gatewayd/actor.rs:123]: I'd track and address the issues with these loops when tackling #1603

[the remark this answers is not in the document]
Comment (okjodom) [reply gateway/ln-gateway/src/gatewayd/actor.rs:123]: At least, that issue is the closest relevant one.

File: fedimint-testing/src/ln/fixtures.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (okjodom) [reply fedimint-testing/src/ln/fixtures.rs]: ack. fixed

File: gateway/ln-gateway/src/gatewayd/actor.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxz42) [reply gateway/ln-gateway/src/gatewayd/actor.rs:123]: ok, :+1:  I added a comment in the issue referencing this second loop :)

**Review by mxz42 — APPROVED** [review]

**Review by justinmoon — APPROVED** [review]

**Pull request merged**